synapse/lib/snap.py

```diff
@@ -894,6 +894,16 @@ async def addNodes(self, nodedefs):
                                         logger.warning(mesg)
                                         continue
 
+                        nodedata = forminfo.get('nodedata')
+                        if nodedata is not None:
+                            for name, data in nodedata.items():
+                                await node.setData(name, data)
```

Comment:
Can either of these (setData) or (addEdge) throw exceptions? This feed function is supposed to be pretty durable to malformed data.

Reply:
I think the only way these could explode would be from pretty radically malformed inputs ( that deserve to blow up )